ChangeLog.d/mbedtls_asn1_type_free.txt

```diff
@@ -0,0 +1,6 @@
+Features
+   * Shared code to free x509 structs like mbedtls_x509_named_data
+
+Removals
```

Comment:
Blocker: `mbedtls_asn1_free_named_data` is part of the API, so it cannot be removed until the next major release. It can only be deprecated.

Reply:
I reverted the change which removes `mbedtls_asn1_free_named_data`.  I have not added deprecation, but will do so in this PR if you recommend.

Reply:
(You're more than welcome to add a commit for the deprecation of `mbedtls_asn1_free_named_data` to avoid the additional round-trip review latency.)

Reply:
:rocket: @gilles-peskine-arm it has been more than 5 weeks since I made the trivial change requested.  Please advise.

Reply:
Sorry, I must have missed the notification here. Pushing back to review.